atox/src/main/kotlin/ToxService.kt

```diff
@@ -124,6 +132,24 @@ class ToxService : LifecycleService() {
                 }
         }
 
+        friendRequestManager.getAll()
+            .filterNotNull()
+            .asLiveData()
+            .observe(this@ToxService) { friendRequests ->
+                if (knownFriendRequests.isEmpty() && friendRequests.isEmpty()) {
+                    return@observe
+                }
+                if (knownFriendRequests.isEmpty()) {
```

Comment:
This won't work if you first get a friend request and then get another. If that happens, knownFriendRequests will only contain the first one.